plugin/src/main/java/io/snyk/languageserver/SnykIssueCache.java

```diff
@@ -0,0 +1,40 @@
+package io.snyk.languageserver;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.snyk.languageserver.protocolextension.messageObjects.scanResults.Issue;
+
+public class SnykIssueCache {
+	private SnykIssueCache() {
+		
+	}
+	
+	private final ConcurrentHashMap<String, Collection<Issue>> snykCodeIssueHashMap = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Collection<Issue>> snykOssIssueHashMap = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Collection<Issue>> snykIaCIssueHashMap = new ConcurrentHashMap<>();
+    
+    private static SnykIssueCache instance = new SnykIssueCache();    
+	public static SnykIssueCache getInstance() {
+		if (instance == null) {
+			synchronized (SnykIssueCache.class) {
+				if (instance == null) {
+					instance = new SnykIssueCache();
+				}
+			}
+		}
+		return instance;
+	}
+    
+    public ConcurrentHashMap<String, Collection<Issue>> getSnykCodeIssueHashMap() {
+		return snykCodeIssueHashMap;
+	}
+	public ConcurrentHashMap<String, Collection<Issue>> getSnykOssIssueHashMap() {
+		return snykOssIssueHashMap;
+	}
+	public ConcurrentHashMap<String, Collection<Issue>> getSnykIaCIssueHashMap() {
+		return snykIaCIssueHashMap;
+	}
```

Comment:
suggestion: we could encapsulate access to the cache maps, so they are not exposed to outside callers. This means, providing methods for adding/removing/getting. If issues are retrieved, `Collections.unmodifiableMap` could be used.